src/administration/administration.service.ts

```diff
@@ -0,0 +1,352 @@
+import {
+  HttpException,
+  HttpStatus,
+  Injectable,
+  InternalServerErrorException,
+  NotFoundException,
+} from '@nestjs/common';
+import { InjectModel } from '@nestjs/mongoose';
+import { Model } from 'mongoose';
+import { AdministrationDto } from './dtos/administration.dto';
+import { StudentDto } from '../student/dtos/student.dto';
+import { JwtService } from '@nestjs/jwt';
+import * as process from 'process';
+
+@Injectable()
+export class AdministrationService {
+  constructor(
+    @InjectModel('Administration') private readonly Administration: Model<any>,
+    @InjectModel('Student') private readonly Student: Model<any>,
+    @InjectModel('Batch') private readonly Department: Model<any>,
+    private readonly jwtService: JwtService,
+  ) {}
+
+  private even: number = 95;
+  private odd: number = 136;
+  async calculateAttendance(student) {
+    let attendance: number = 0;
+
+    if (parseInt(student.currentSem) % 2 === 0) {
+      attendance = (student.attendance.length / this.even) * 100;
+    }
+    if (parseInt(student.currentSem) % 2 === 1) {
+      attendance = (student.attendance.length / this.odd) * 100;
+    }
+
+    return attendance.toFixed(2);
+  }
+
+  async createAdmin(Admin: AdministrationDto) {
+    if (await this.Administration.findOne({ profile: 'admin' })) {
+      throw new HttpException('Admin already exists', HttpStatus.CONFLICT);
+    }
+    const admin = new this.Administration(Admin);
+    await admin.save();
+    return admin;
+  }
+
+  async addStaff(Staff: AdministrationDto) {
+    if (
+      await this.Administration.findOne({
+        administratorId: Staff.administratorId,
+      })
+    ) {
+      throw new HttpException('Staff already exists', HttpStatus.CONFLICT);
+    }
+    const staff = new this.Administration(Staff);
+    await staff.save();
+    return 'staff added Successfully';
+  }
+
+  async addStudent(studentData: StudentDto) {
+    if (
+      await this.Student.findOne({
+        studentId: studentData.studentId,
+      })
+    ) {
+      throw new HttpException('Student already exists', HttpStatus.CONFLICT);
+    }
+    const student = new this.Student(studentData);
+    await student.save();
+    return 'Student added Successfully';
+  }
+
+  async login(administratorId: string, password: string) {
+    const administrator: any = await this.Administration.findOne({
+      administratorId,
+    });
+    if (!administrator) {
+      throw new HttpException('User not found', HttpStatus.NOT_FOUND);
+    }
+    const match = password === administrator.password;
+    if (match) {
```

Comment:
Here you can implement early return funcationlity like, you check reverse condition